webapp/src/pages/dashboard/account/history.tsx

```diff
@@ -150,64 +147,81 @@ export default function AccountHistory() {
                   justifyContent="space-between"
                   mt={4}
                 >
-                  <Flex alignItems="center" gap={2}>
-                    <Box
-                      borderRadius="2xl"
-                      flexShrink={0}
-                      bgColor={darkenPartnerColor}
-                      p={1.5}
-                    >
-                      <Image
-                        src={userHistoryItem.offer.partner.icon.url as string}
-                        alt={userHistoryItem.offer.partner.icon.alt as string}
-                        width={36}
-                        height={36}
-                        style={{ borderRadius: "8px" }}
-                      />
-                    </Box>
-                    <Flex flexDir="column" justifyContent="start" w="full">
-                      <Text fontSize={14} fontWeight={500}>
-                        {userHistoryItem.offer.partner.name}
-                      </Text>
-                      <Text fontSize={12} fontWeight={500} noOfLines={1}>
-                        {userHistoryItem.offer.title}
-                      </Text>
-                      <Box mt={1}>
-                        {"used" in userHistoryItem && !userHistoryItem.used ? (
-                          <Flex alignItems="center" color="disabled" gap={0.5}>
-                            <Icon as={HiMiniClock} w={3} h={3} />
-                            <Text fontWeight={500} fontSize={12}>
-                              Fin{" "}
-                              {currentCouponUsedAt.toLocaleDateString("fr-FR", {
-                                day: "2-digit",
-                                month: "2-digit",
-                              })}
-                            </Text>
-                          </Flex>
-                        ) : (
-                          <Flex
-                            alignItems="center"
-                            fontSize={12}
-                            px={1}
-                            color="success"
-                            bgColor="successLight"
-                            w="fit-content"
-                            borderRadius="2.5xl"
-                            gap={0.5}
-                            fontWeight={500}
-                          >
-                            <Icon
-                              as={HiMiniCheckCircle}
-                              stroke="white"
-                              w={3}
-                              h={3}
-                            />
-                            Déjà utilisée
-                          </Flex>
-                        )}
+                  <ConditionalLink
+                    condition={"status" in userHistoryItem}
```

Comment:
peut être met `"number" in`
car si on décide de mettre un status sur les coupons on va pas ce rappeler de ça :grimacing: